src/mesh/RadioLibInterface.cpp

```diff
@@ -274,23 +274,39 @@ void RadioLibInterface::onNotify(uint32_t notification)
             if (!canSendImmediately()) {
                 setTransmitDelay(); // currently Rx/Tx-ing: reset random delay
             } else {
-                if (isChannelActive()) { // check if there is currently a LoRa packet on the channel
-                    startReceive();      // try receiving this packet, afterwards we'll be trying to transmit again
-                    setTransmitDelay();
+                meshtastic_MeshPacket *txp = txQueue.getFront();
+                assert(txp);
+                // check for any remaining delay, and abort if we're still supposed to be waiting
+                if (txp->tx_after > millis()) {
+                    // There's still some delay pending on this packet, so resume waiting for it to elapse
+                    notifyLater(txp->tx_after - millis(), TRANSMIT_DELAY_COMPLETED, false);
                 } else {
-                    // Send any outgoing packets we have ready as fast as possible to keep the time between channel scan and
-                    // actual transmission as short as possible
-                    meshtastic_MeshPacket *txp = txQueue.dequeue();
-                    assert(txp);
-                    bool sent = startSend(txp);
-                    if (sent) {
-                        // Packet has been sent, count it toward our TX airtime utilization.
-                        uint32_t xmitMsec = getPacketTime(txp);
-                        airTime->logAirtime(TX_LOG, xmitMsec);
+                    if (isChannelActive()) { // check if there is currently a LoRa packet on the channel
+                        startReceive();      // try receiving this packet, afterwards we'll be trying to transmit again
+                        if (txp->tx_after) {
+                            // If we're using the late rebroadcast window, kick it down the road some more
+                            txp->tx_after = max(txp->tx_after, (uint32_t)millis()) +
+                                            (txp->rx_snr ? getTxDelayMsecWeighted(txp->rx_snr) : getTxDelayMsec());
```

Comment:
A value of 0 is valid for `rx_snr`, it’s better to check for `rx_rssi`.

Reply:
Resolved in setTransmitDelay().